x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/base_validator.ts

```diff
@@ -74,6 +77,22 @@ export class BaseValidator {
     }
   }
 
+  protected async validateHasEndpointExceptionsPrivileges(
+    privilege: keyof EndpointAuthz
+  ): Promise<void> {
+    const appFeaturesService = this.endpointAppContext.getAppFeaturesService();
+
+    const isEndpointExceptionsFeaturesEnabled = (await appFeaturesService).isEnabled(
+      AppFeatureSecurityKey.endpointExceptions
+    );
+
+    // for serverless only when Endpoint Exceptions is not enabled
+    // verify that the user has the privilege to create/read/update/delete endpoint exceptions
+    if (!isEndpointExceptionsFeaturesEnabled && !(await this.endpointAuthzPromise)[privilege]) {
+      throw new EndpointExceptionsValidationError('Endpoint exceptions authorization failure', 403);
+    }
```

Comment:
Not really a fan of this should being done here. It can lead to these types of logic changes happening in multiple places.

It would be best if the Authz for exceptions (`canReadEndpointExceptions`, `canWriteEndpointExceptions`) were defined in a way that they already took into consideration the appFeature (just like we do today when we check Kbn license levels).  Maybe `appFeatues` should be pass into out `authz.ts` during calculation?

Reply:
Yes, this conditional is not correct, we should only check the capabilities here: 

```
if (!(await this.endpointAuthzPromise)[privilege]) {
```

As per the configuration of the `endpointExceptions` AppFeatureKey (in [ess](https://github.com/elastic/kibana/blob/1f7a383b7becc94d8a13b8b2163635f3bf204d02/x-pack/plugins/security_solution_ess/server/app_features/security_app_features_config.ts#L48-L50) and in [serverless](https://github.com/elastic/kibana/blob/1f7a383b7becc94d8a13b8b2163635f3bf204d02/x-pack/plugins/security_solution_serverless/server/app_features/security_app_features_config.ts#L41-L43)), we can check those capabilities (via endpoint authz module in your case), and that's all we need to do: 

- If the `AppFeatureSecurityKey.endpointExceptions` is enabled: the _showEndpointExceptions_ and _crudEndpointExceptions_ capabilities will be added to the Security Solution privileges, and they will be granted depending on the user role.

- If the `AppFeatureSecurityKey.endpointExceptions` is not enabled: the _showEndpointExceptions_ and _crudEndpointExceptions_ capabilities will never be added to the Security Solution privileges, so they will never be granted, regardless of the user role.

@paul-tavares In the `authz.ts`, the appFeatures keys are already taken into consideration when checking the capabilities, it works in a declarative way using AppFeaturesConfigs linked above, there's no need to do explicit checks to the appFeatures in the authz code.

When validating privileges, as a general rule of thumb to not overcomplicate things, we can ask ourselves:

Do we have access to the authz module (or Kibana capabilities directly)?
    - Yes: Check them, that's all we need to do.
    - No: Get the AppFeaturesService and check if the key is enabled (knowing that this method will not take role privileges into account)

Reply:
02eb0a28ea613c5d308852eb926176f2b401b91a